### ISSUE TITLE
Setup Dimentions failure

### ISSUE THREAD

**Issue by Trident35ro** [popstarfreas/Dimensions#122]
Some people got it work but I cant. Used the commands said in the instructions: `npm install --only=production` and `npm run start` in the folder with the Dimentions files. After 
running these commands and got this output:

> @popstarfreas/dimensions@5.10.0-beta.1 start
> cd ./build && node ./index.js

I expected to get this output:

> @popstarfreas/dimensions@5.10.0-beta.1 start
> cd ./build && node ./index.js
> 
> [ _any 4 digit number idk_ ] Server on _port 1_ started
> [ _any 4 digit number idk_ ] Server on _port 2_ started <

Thinking that maybe it shouldn’t show me that, I joined the server and tried commands like `/dimensions` , `/ _server name in config_ ` and even tried with `/sudo` . Nothing worked! Can somebody help me?

**Comment by popstarfreas** [comment]
What does your config look like?

**Comment by Trident35ro** [comment]
`"use strict"
exports.ConfigSettings = {
    servers: [
        {
            listenPort: 9000,
            routingServers: [{
                name: "lobby",
                serverIP: " _server ip_ ",
                serverPort: 9000,
                hidden: false,
            }]    
        },
        {
            routingServers: [{
                name: "world",
                serverIP: " _server ip_ ",
                serverPort: 9001,
                hidden: false,
            }]    
        },
    ],
}`

**Comment by Trident35ro** [comment]
By the way, where should the Dimensions files be? Like in the plugin folder in tShock or in any location.

**Comment by popstarfreas** [comment]
Dimensions is its own program so it doesn't need to be in the tshock folder, only the actual dimensions dll should be in the plugins folder.

With that config I think all logging is disabled that could be why you don't see any messages. It looks like you set listen port and server port to 9000. They should be different. Listen port is what you will use to connect and server port is what port your tshock server is using.

**Comment by Trident35ro** [comment]
I changed the listen port to one of the server ports because I thought it would work, and if I joined the server using the port 9000, it wouldn't redirect me to "lobby." It sounds stupid, but it was worth trying. Maybe this would help, I use Dimensions 5.10, but tried a little bit on 6.0.0.

**Comment by popstarfreas** [comment]
The latest version is the best though I've not built with all the recent changes yet. What happens when you change the listen port to something else and join using that port?

**Comment by Trident35ro** [comment]
Nothing. Maybe the problem is that I don't know how to setup the plugin.

**Comment by popstarfreas** [comment]
What do you mean by nothing? When you use your game to join that port what happens?

**Comment by Trident35ro** [comment]
Nothing different happens

**Comment by popstarfreas** [comment]
You get connected to the server still?

**Comment by popstarfreas** [comment]
Perhaps the example needs to be improved if it's not good enough

**Comment by Trident35ro** [comment]
> You get connected to the server still?

If I use the listen port, it won't.

**Comment by popstarfreas** [comment]
It gets stuck on connecting or a different message?

**Comment by Trident35ro** [comment]
No! It says "Timeout connection", like the server doesn't exist

**Comment by Trident35ro** [comment]
And your plugin works with tShock servers hosted on pcs or laptops?

**Comment by popstarfreas** [comment]
The plugin itself should, yes. But the plugin only rewrites players' IP addresses to be correct when they connect through dimensions.

**Comment by Trident35ro** [comment]
I think that the plugin doesn't even know that my server exists. But can you sent me an screenshot that shows what the cmd window should output after running `npm run start`?

**Comment by pacenadhif777** [comment]
> I think that the plugin doesn't even know that my server exists. But can you sent me an screenshot that shows what the cmd window should output after running `npm run start`?

i have same problem too

**Comment by popstarfreas** [comment]
Are you still having this issue?

**Comment by Trident35ro** [comment]
I stopped making a tShock server and deleted it completely so ill respond the resurt when I finish configuring it again.

**Comment by Trident35ro** [comment]
I made the server but still have the same problem.

> Dimensions is its own program so it doesn't need to be in the tshock folder, only the actual dimensions dll should be in the plugins folder.

I looked back at this message and I realised I dont have any dll file. Should it be in the dimensions folder?

**Comment by popstarfreas** [comment]
The DLL is just the server plugin: https://github.com/popstarfreas/Dimensions-TerrariaServer/releases it's not needed for functionality to work, it's there to allow the tshock server to update the client IP to be correct. What's your current config?

**Comment by Trident35ro** [comment]
> What's your current config?

https://pastebin.pl/view/0b3a3f02
I will try with the dll to see if it does something. And I see in the config there is a line that says Terraria version to be 1.4.3.6.

**Comment by Trident35ro** [comment]
Update: I changed the version in config and now it finally works. And yea, the tutorial should be updated to say this, also to mention you need to install Git for building the plugin.

**Comment by popstarfreas** [comment]
If you're talking about the rest api config that shouldn't affect players joining your server.

**Comment by Trident35ro** [comment]
Ok, but in the installation guide should mention that you need Git installed.
